src/webgpu/api/operation/adapter/requestDevice.spec.ts

```diff
@@ -78,6 +78,72 @@ g.test('invalid')
     t.expect(lost.reason === undefined);
   });
 
+g.test('stale')
+  .desc(
+    `
+    Test that adapter.requestDevice() can successfully return a device once, and once only.
+    - Tests that we can successfully resolve after serial and concurrent rejections.
+    - Tests that consecutive valid attempts only succeeds the first time, returning lost device otherwise.`
+  )
+  .paramsSubcasesOnly(u =>
+    u
+      .combine('initialError', [undefined, 'TypeError', 'OperationError'])
+      .combine('awaitInitialError', [true, false])
+      .combine('awaitSuccess', [true, false])
+      .filter(({ initialError, awaitInitialError }) => {
+        return initialError !== undefined && !awaitInitialError;
+      })
+  )
+  .fn(async t => {
+    const gpu = getGPU();
+    const adapter = await gpu.requestAdapter();
+    assert(adapter !== null);
+
+    const { initialError, awaitInitialError, awaitSuccess } = t.params;
+
+    switch (initialError) {
+      case undefined:
+        break;
+      case 'TypeError':
+        // Cause a type error by requesting with an unknown feature.
+        if (awaitInitialError) {
+          await assertReject(
+            adapter.requestDevice({ requiredFeatures: ['unknown-feature' as GPUFeatureName] })
+          );
+        } else {
+          t.shouldReject(
+            'TypeError',
+            adapter.requestDevice({ requiredFeatures: ['unknown-feature' as GPUFeatureName] })
+          );
+        }
+        break;
+      case 'OperationError':
+        // Cause an operation error by requesting with an unknown limit.
+        if (awaitInitialError) {
+          await assertReject(adapter.requestDevice({ requiredLimits: { unknownLimitName: 9000 } }));
+        } else {
+          t.shouldReject(
+            'OperationError',
+            adapter.requestDevice({ requiredLimits: { unknownLimitName: 9000 } })
+          );
+        }
+        break;
+    }
+
+    const promise = adapter.requestDevice();
+    if (awaitSuccess) {
+      const device = await promise;
+      assert(device !== null);
```

Comment:
nit: make sure the devices gets destroyed at the end of the test

Reply:
Done.